diff --git a/Target/Source/ARMCM3_STM32F1/can.c b/Target/Source/ARMCM3_STM32F1/can.c
index 4509a89f..01d30e01 100644
--- a/Target/Source/ARMCM3_STM32F1/can.c
+++ b/Target/Source/ARMCM3_STM32F1/can.c
@@ -234,7 +234,7 @@ static blt_bool CanGetSpeedConfig(blt_int16u baud, blt_int16u *prescaler,
 void CanInit(void)
 {
   blt_int16u prescaler;
-  blt_int8u  tseg1, tseg2;
+  blt_int8u  tseg1 = 0, tseg2 = 0;
   blt_bool   result;
   blt_int32u timeout;
   blt_int32u rxMsgId = BOOT_COM_CAN_RX_MSG_ID;
